tests/lib/rook/upstream.py

```diff
@@ -28,81 +29,88 @@ class RookCluster(RookBase):
-                bar=None,
-            )
-
-            logger.info("[build_rook] Unpack go")
-            execute(
-                "tar -C %s -xzf %s"
-                % (self.builddir,
-                   os.path.join(self.builddir, 'go-amd64.tar.gz'))
-            )
-
-            logger.info("[build_rook] Make rook")
-            execute(
-                "PATH={builddir}/go/bin:$PATH GOPATH={builddir} "
-                "TMPDIR={tmpdir} "
-                "make --directory='{builddir}/src/github.com/rook/rook' "
-                "-j BUILD_REGISTRY='rook-build' IMAGES='ceph' "
-                "build".format(builddir=self.builddir,
-                               tmpdir=self.go_tmpdir),
+        self.get_rook()
+        if not converter('@bool', settings.UPSTREAM_ROOK.BUILD_ROOK_FROM_GIT):
+            return
+
+        self.get_golang()
+        logger.info("Compiling rook...")
+        execute(command="make -j BUILD_REGISTRY='rook-build' IMAGES='ceph'",
+                env={"PATH": f"{self.workspace.bin_dir}/go/bin:"
+                             f"{os.environ['PATH']}",
+                     "TMPDIR": self.workspace.tmp_dir,
+                     "GOCACHE": self.workspace.tmp_dir,
+                     "GOPATH": self.workspace.build_dir},
                 log_stderr=False,
                 logger_name=(
-                    "make -j BUILD_REGISTRY='rook-build' IMAGES='ceph'"),
-            )
-
-            logger.info("[build_rook] Tag image")
-            execute('docker tag "rook-build/ceph-amd64" rook/ceph:master')
+                    "make -j BUILD_REGISTRY='rook-build' IMAGES='ceph'"))
+
+        image = 'rook/ceph'
+        tag = f"{settings.UPSTREAM_ROOK.VERSION}-rookcheck"
+        self.rook_image = f"{image}:{tag}"
+        logger.info(f"Tag image as {image}:{tag}")
+        execute(f'docker tag "rook-build/ceph-amd64" {image}:{tag}')
+
+        logger.info("Save image tar")
+        # TODO(jhesketh): build arch may differ
+        execute(f"docker save {image}:{tag} | gzip > %s"
+                % os.path.join(self.build_dir, 'rook-ceph.tar.gz'))
+        self._rook_built = True
 
-            logger.info("[build_rook] Save image tar")
-            # TODO(jhesketh): build arch may differ
-            execute(
-                "docker save rook/ceph:master | gzip > %s"
-                % os.path.join(self.builddir, 'rook-ceph.tar.gz')
-            )
+    def upload_rook_image(self):
+        self.kubernetes.hardware.ansible_run_playbook(
+            "playbook_rook_upstream.yaml")
 
-        self.ceph_dir = os.path.join(
-            self.builddir,
-            'src/github.com/rook/rook/cluster/examples/kubernetes/ceph'
+    def get_rook(self):
+        logger.info("Clone rook repo %s" % settings.UPSTREAM_ROOK.REPO)
+        execute(
+            "git clone %s %s" % (settings.UPSTREAM_ROOK.REPO, self.build_dir),
+            log_stderr=False
+        )
+        os.chdir(self.build_dir)
```

Comment:
This can have some difficult side effects. For example, the collection path for the logs at the end the user expects to be relative to where they are running rookcheck, but if we change at some point during the process we will end up in the wrong place. Instead, pushd into the dir as part of the `execute` command.

Reply:
I just updated to checkout the right branch while cloning.